### ISSUE TITLE
Specifying registry mirror for "*" results in: Task failed, error was: runtime: unmarshalling manifest "webhook.yaml"

### ISSUE THREAD

**Issue by judge-red** [kubermatic/kubeone#2926]
### What happened?

If specifying `.containerRuntime.containerd.registries."*"` (with appropriate childs), applying the `machine-controller` addon will fail:

```
INFO[16:55:44 CEST] Applying addon coredns-pdb...                
INFO[16:55:46 CEST] Applying addon metrics-server...             
INFO[16:55:49 CEST] Applying addon cni-cilium...                 
INFO[16:55:53 CEST] Applying addon nodelocaldns...               
INFO[16:55:56 CEST] Applying addon machinecontroller...          
WARN[16:55:56 CEST] Task failed, error was: runtime: unmarshalling manifest "webhook.yaml"
error converting YAML to JSON: yaml: line 8: did not find expected alphabetic or numeric character 
WARN[16:56:06 CEST] Retrying task...                             
INFO[16:56:06 CEST] Applying addon coredns-pdb...                
INFO[16:56:09 CEST] Applying addon metrics-server...             
INFO[16:56:11 CEST] Applying addon cni-cilium...                 
INFO[16:56:15 CEST] Applying addon nodelocaldns...               
INFO[16:56:18 CEST] Applying addon machinecontroller...          
WARN[16:56:18 CEST] Task failed, error was: runtime: unmarshalling manifest "webhook.yaml"
error converting YAML to JSON: yaml: line 8: did not find expected alphabetic or numeric character 
WARN[16:56:32 CEST] Retrying task...                     
```

If I replace `"*"` with e.g. `"docker.io"`, it works as expected. However, `"*"` is explicitely supported by containerd, and is rather useful.

It should be noted, that containerd on the control plane nodes is correctly configured with `"*"`, the error occurs afterwards.

### Expected behavior

No error :)

### How to reproduce the issue?

Add something like the below to your `kubeone.yaml`:

```
containerRuntime:
  containerd:
    registries:
      "*":
        mirrors:
          - https://my-cr.tld
```

### What KubeOne version are you using?

<details>

```console
$ kubeone version
{
  "kubeone": {
    "major": "1",
    "minor": "7",
    "gitVersion": "1.7.0",
    "gitCommit": "1195366fd0cf11f314d194a3b29b6a782afde9a8",
    "gitTreeState": "",
    "buildDate": "2023-09-08T14:02:33Z",
    "goVersion": "go1.20.5",
    "compiler": "gc",
    "platform": "darwin/amd64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "57",
    "gitVersion": "v1.57.3",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

</details>

### Provide your KubeOneCluster manifest here (if applicable)

Let me know if the above snippet isn't enough but pretty sure the rest doesn't matter.

<details>

```yaml
# paste manifest here
```

</details>

### What cloud provider are you running on?

OpenStack

### What operating system are you running in your cluster?

Ubuntu 22.04

### Additional information

I'm kinda thinking this is a regression, i.e. I think I tried using registry mirrors like that before (but then didn't use the feature at all because something else was missing - only retried now, probably several months later).

**Comment by judge-red** [comment]
@kron4eg Thanks for the quick fix - can we get this into 1.7.1 (after/if it's merged into main), please?

**Comment by xmudrii** [comment]
@judge-red Yes, cherry-picks for v1.7 and v1.6 are already created. :)
